src/Protocol/Protocol_1_8.cpp

```diff
@@ -1420,18 +1420,26 @@ void cProtocol_1_8_0::SendSpawnMob(const cMonster & a_Mob)
 void cProtocol_1_8_0::SendStatistics(const cStatManager & a_Manager)
 {
 	ASSERT(m_State == 3);  // In game mode?
-
+	std::vector<Statistic> send = {};
```

Comment:
We're allocating a vector then throwing it away almost immediately; there's gotta be a better way.

If we went with GetDirtyStatistics returning the dirty array, then the first pass could unset the ones which our protocol don't support, and count the rest. We then write the size, and the second pass would write the data for the dirties remaining.

Also on style: prefer captialised `Send`

Reply:
Your idea is quite good

Style is gonna be changed

Reply:
Oh - I just noticed - since the a_Manager is constant in the transmitting we can't write into the map :/